spec/person_spec.rb

```diff
@@ -0,0 +1,35 @@
+require_relative '../person'
+
+describe Person do
+  before :each do
+    @person = Person.new(30, 'Shubham', 'true')
+  end
+
+  it 'checking person instance' do
+    expect(@person).to be_instance_of Person
+  end
+
+  it 'checking attributes' do
+    expect(@person).to have_attributes(age: 30, name: 'Shubham', parent_permission: 'true')
+  end
+
+  it 'test for add_rental' do
+    book = double('book', rentals: [])
+    allow(book).to receive(:title) { 'Harry Potter' }
+    allow(book).to receive(:author) { 'Shubham' }
+    rental = @person.add_rental('2323-23-23', book)
+    expect(rental.book.title).to eq('Harry Potter')
+  end
+
+  it 'correct name method' do
+    expect(@person.correct_name).to eq('Shubham')
+  end
+
+  it 'correct age method' do
+    expect(@person.send('of_age?')).to be true
+  end
+
+  it 'can use services' do
+    expect(@person.can_use_services?).to be true
+  end
+end
```

Comment:
- [ ] You should test the scenario of the constructor method sets a default value for the optional name parameter if it is not provided 🙃 
- [ ] Also the scenario of the `can_use_services?` method returns false if under age and no parent permission 😁 
- [ ] You should test also the scenario of the `can_use_services?` method returns true if has parent permission 😁

Reply:
Thanks! @cynthiainga, more tests have been added.